lib/smci_report.rb

```diff
@@ -0,0 +1,220 @@
+            logger.error "too many publications found for this author in WoS, query shown below.  Pubs found: #{num_pubs_returned} > #{Settings.WOS.max_publications_per_author}"
+            logger.info author_query.send(:author_query) # log the query being sent to WoS for debugging
+          end
+          # end check for too many publications for this author
+        end
+        # end check if this was a profiles author or not
+      end
+      # end loop over all authors in the CSV
+    end
+    # end writing the output CSV
+
+    end_time = Time.zone.now
+    logger.info ''
+    logger.info "Total publications exported: #{total_pubs}. Output file: #{@output_file}. Total time: #{((end_time - start_time) / 60.0).round(1)} minutes."
+    logger.info 'Completed export.'
+    logger.info '*****************'
+  end
+
+  private
+
+  def header_row
+    %w(title author_list pmid wos_uid sul_pub_id doi doi_url publisher journal volume articlenumber supplement issue pages mesh pub_year pub_date provenance profiles_author_last_name profiles_author_first_name profiles_author_sunet profiles_author_cap_profile_id profiles_author_employee_id profiles_author_email pub_harvested_date apa_citation mla_citation chicago_citation)
+  end
+
+  def output_row(pub_hash, author, harvested_at)
+    author_list = pub_hash[:author] ? Csl::RoleMapper.send(:parse_authors, pub_hash[:author]).map { |a| "#{a['family']}, #{a['given']}" }.join('; ') : ''
+    pmid = pub_hash[:identifier].map { |ident| ident[:id] if ident[:type].downcase == 'pmid' }.compact.join('')
+    doi = pub_hash[:identifier].map { |ident| ident[:id] if ident[:type].downcase == 'doi' }.compact.join('')
+    wos_uid = pub_hash[:identifier].map { |ident| ident[:id] if ident[:type].downcase == 'WosUID' }.compact.join('')
+    sul_pub_id = pub_hash[:identifier].map { |ident| ident[:id] if ident[:type].downcase == 'SULPubId' }.compact.join('')
+    doi_url = pub_hash[:identifier].map { |ident| ident[:url] if ident[:type].downcase == 'doi' }.compact.join('')
+    journal = pub_hash[:journal] ? pub_hash[:journal][:name] : ''
+    issue = pub_hash[:issue] ? pub_hash[:journal][:issue] : ''
+    volume = pub_hash[:journal] ? pub_hash[:journal][:volume] : ''
+    article_number = pub_hash[:journal] ? pub_hash[:journal][:articlenumber] : ''
+    mesh = pub_hash[:mesh_headings] ? pub_hash[:mesh_headings].map { |h| h[:descriptor][0][:name] }.compact.reject(&:empty?).join('; ') : ''
+    if author # if this is a profiles author, then we will output this info as well
+      author_last_name = author.last_name
+      author_first_name = author.first_name
+      author_sunet = author.sunetid
+      author_cap_profile_id = author.cap_profile_id
+      author_univ_id = author.university_id
+      author_email = author.email
+    else
+      author_last_name = author_first_name = author_sunet = author_cap_profile_id = author_univ_id = author_email = ""
```

Comment:
Can you add a test for this branch?

Reply:
added a few more tests to hit this branch of the code